File: benchmarks/pandas/bench_fillna_dropna.py
import pandas as pd, time, json
N = 100_000
data = [None if i % 7 == 0 else i * 1.5 for i in range(N)]
s = pd.Series(data)
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    s.fillna(0)
    s.dropna()
t0 = time.perf_counter()
for _ in range(ITERS):
    s.fillna(0)
    s.dropna()
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "fillna_dropna", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))

File: benchmarks/tsb/bench_fillna_dropna.ts
import { fillna, dropna } from "tsb";
import { Series } from "tsb";
const N = 100_000;
const data: (number | null)[] = Array.from({ length: N }, (_, i) => (i % 7 === 0 ? null : i * 1.5));
const s = new Series(data);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) {
  fillna(s, { value: 0 });
  dropna(s);
}
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) {
  fillna(s, { value: 0 });
  dropna(s);
}
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "fillna_dropna", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));
